staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go

```diff
@@ -21,9 +21,22 @@ import (
 	"time"
 )
 
-// QueueSetFactory is used to create QueueSet objects.
+// QueueSetFactory is used to create QueueSet objects.  Creation, like
+// config update, is done in two phases: the first phase consumes the
+// QueuingConfig and the second consumes the DispatchingConfig.  They
+// are separated so that errors from the first phase can be found
+// before committing to a concurrency allotment for the second.
 type QueueSetFactory interface {
-	NewQueueSet(config QueueSetConfig) (QueueSet, error)
+	// QualifyQueuingConfig does the first phase of creating a QueueSet
+	QualifyQueuingConfig(QueuingConfig) (QueueSetCompleter, error)
+}
+
+// QueueSetCompleter finishes the two-step process of creating or
+// reconfiguring a QueueSet
+type QueueSetCompleter interface {
+	// GetQueueSet returns a QueueSet configured by the given
+	// dispatching configuration.
+	GetQueueSet(DispatchingConfig) QueueSet
```

Comment:
Maybe `Apply`? `GetQueueSet` sounds like a search, but this is a mutation (or construction) if I am reading it right.

Reply:
How about just `Complete`?

Reply:
Yes, "Complete" would be consistent with the interface name.